src/core/Action.js

```diff
@@ -216,6 +216,25 @@ export default class Action extends EventEmitter {
       id: id,
       action: this.name,
       values: result,
+      status: 4,  // Corresponds to GoalStatus.STATUS_SUCCEEDED
```

Comment:
Can we extract GoalStatus to an enum, since we've got typescript?

Reply:
Apparently since these are `.js` files, things are not happy if I try it directly:

```
src/core/Action.js:14:6 - error TS8006: 'enum' declarations can only be used in TypeScript files.
```

Is there some easy way around this that fits within this PR?

Reply:
I found this... please let me know if there's anything better.

https://stackoverflow.com/a/44447975

Reply:
Did you try adding a typescript file for the enum and importing it into the JS file? I don't know in practice how that works, but it should transpile into an object?

Reply:
Yeaaaa, that did work! Just had to add one more compiler option. Thank you.